cli/azd/cmd/deploy.go

```diff
@@ -152,18 +152,24 @@ func (d *deployAction) Run(ctx context.Context, cmd *cobra.Command, args []strin
 		}
 
 		if interactive {
-			deployMsg := fmt.Sprintf("Deploying service %s", svc.Config.Name)
-			fmt.Println(deployMsg)
-			spinner := spin.NewSpinner(deployMsg)
+			deployMsg := fmt.Sprintf("Deploying service %s...", output.WithHighLightFormat(svc.Config.Name))
+			console.Message(ctx, deployMsg)
+
+			spinner := spin.GetSpinner(ctx)
+			if spinner == nil {
+				spinner = spin.NewSpinner(deployMsg)
+				ctx = spin.WithSpinner(ctx, spinner)
+			}
```

Comment:
Do we have two components that are both trying to update the same spinner at the same time? I wonder if it's possible for us to have the UI be managed by a single component, i.e. the `cmd`, and downstream components should simply be sending back messages back to `cmd` that affects the spinner. It seems confusing to have multiple components all trying to create a spinner at the same time.

The current proposal seems to be moving towards a singleton solution. I think this might bite us later in the case where we have multiple concurrent components all trying to do conflicting operations, but perhaps we can be somewhat careful in avoiding such interactions.

Reply:
We have one case today in the container app deploy command.  Deploy leverages Bicep to push the new container image reference.  We will need to update this in the near future to revert back to az cli commands.

They are not performing conflicting operations - just reporting status message on the spinner.

Agreed that we will likely need to consider how to address this long term.